File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main(void)
{
    stack<int> st;
    int n;
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        string cmd;
        cin >> cmd;
        if (cmd == "push")
        {
            int s;
            cin >> s;
            st.push(s);
        }
        else if (cmd == "pop")
        {
            if (st.empty())
                cout << "-1" << endl;
            else
            {
                cout << st.top() << endl; 
                st.pop();
            }     
        }
        else if (cmd == "size")
        {
            cout << st.size() << endl;
        }
        else if (cmd == "empty")
        {
            if (st.empty())
                cout << "1" << endl;
            else
                cout << "0" << endl;
        }
        else if (cmd == "top")
        {
            if (st.empty())
                cout << "-1" << endl;
            else
                cout << st.top() << endl;
        }
    }
}

File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

n = int(input())
stack = []
output = []

for i in range(n):
    order = input()
    if order.startswith('push'):
        stack.append(order.split()[1])
    elif order.startswith('pop'):
        if stack:
            output.append(stack.pop())
        else:
            output.append('-1')
    elif order.startswith('size'):
        output.append(str(len(stack)))
    elif order.startswith('empty'):
        output.append('0' if stack else '1')
    elif order.startswith('top'):
        output.append(stack[-1] if stack else '-1')
        
print('\n'.join(output))
        
